docusaurus/docs/reactnative/guides/auto_complete_suggestion_list_customizations.mdx

```diff
@@ -0,0 +1,164 @@
+        }
+    }}>
+    {/*The underlying components*/}
+</Channel>
+```
+
+Note: The default `AutoCompleteSuggestionItem` must be returned if you don't want any change for a particular trigger type.
+
+### Customizing the AutoComplete suggestion list
+
+The list can also be entirely customized with its own header, item and other definitions. This can be done using the prop `AutoCompleteSuggestionList` in the Channel component.
+
+The props available to the component are:
+- `active`: Checks the list is active or not.
+- `headerProps`: The props for the header of the list. This can take `type` and `value` as explained above in the header section.
+- `suggestions`: The suggestions array which contains the data for the list to be displayed in the list items.
+- `type`: The trigger type of the suggestion list.
+
+An example to demonstrate the same is as follows. The different ways of usage is also explained:
+
+```jsx
+import { AutoCompleteSuggestionHeader, AutoCompleteSuggestionItem } from 'stream-chat-react-native';
+import { FlatList, Text, View } from 'react-native';
+
+<Channel
+    AutoCompleteSuggestionList={({
+        headerProps: { value },
+        suggestions: { query, data, onSelect },
+        type,
+    }) => {
+        if (type === 'command') {
+            return (
+                <View>
+                    <View>
+                        <Text>Command suggestions for {value}</Text>
+                    </View>
+                    {data.map((item) => (
+                        <AutoCompleteSuggestionItem itemProps={item} type={type} />
+                    ))}
+                </View>
+            );
+        } else if (type === 'emoji') {
+            return (
+                <FlatList
+                    data={data}
+                    keyboardShouldPersistTaps='always'
+                    ListHeaderComponent={<AutoCompleteSuggestionHeader type={type} value={value} />}
+                    renderItem={({ index, item }) => {
+                        return (
+                            <TouchableOpacity
+                                onPress={() => {
+                                onSelect(item);
+                                }}
+                            >
+                                <Text>{item.name}</Text>
+                            </TouchableOpacity>
+                        );
+                    }}
+                />
+            );
+        } else {
+            return (
+                <View>
+                    <AutoCompleteSuggestionHeader type={type} value={query} />
+                    {data.map((item) => (
+                        <AutoCompleteSuggestionItem itemProps={item} type={type} />
+                    ))}
+                </View>
+            );
+        }
+    }}>
+    {/*The underlying components*/}
+</Channel>
+```
+
+Note: The default `AutoCompleteSuggestionList` must be returned if you don't want any change for a particular trigger type.
```

Comment:
````suggestion
You can customize the AutoCompleteSuggestionList and its child components by passing customized in the following props on a Channel:

- AutoCompleteSuggestionHeader
- AutoCompleteSuggestionItem
- AutoCompleteSuggestionList

The AutoCompleteSuggestionList provided out-of-the-box will render a FlatList using AutoCompleteSuggestionHeader as the list's header and render an AutoCompleteSuggestionItem for each item in the list.

You can choose to conditionally render custom components for selected trigger types and keep the built-in implementation for others, or only render customized components for all trigger types by checking the string passed as a prop to AutoCompleteSuggestionList.

The available suggestion trigger types are selected by a symbol followed by text, and are as follows:

| Trigger symbol | Trigger type |
|----------------|--------------|
| `/`            | `"command"`  |
| `:`            | `"emoji"`    |
| `@`            | `"mention"`  |

### Customizing Header component

The AutoCompleteSuggestionHeader component is rendered at the top of the FlatList in AutoCompleteSuggestionList. You can customize this by passing your own component to the `AutoCompleteSuggestionHeader` prop on a Channel.

The props your component will receive are as defined by the `AutoCompleteSuggestionHeaderProps` type:

- `type`: The trigger type used to open the suggestion list
- `value`: The query text for the suggestion

For example, given the trigger `@BruceBanner`, the @ will give the trigger type `"mention"`, and a query text value of `"BruceBanner"`.

An example passing a custom AutoCompleteSuggestionHeader that renders different components for `command` and `emoji`, but keeps the default implementation for `mention` suggestions:

```jsx
import { Text } from 'react-native';
import {
  AutoCompleteSuggestionHeader,
  AutoCompleteSuggestionHeaderProps,
  Channel,
} from 'stream-chat-react-native';

const MyCustomHeader = (props: AutoCompleteSuggestionHeaderProps) => {
  const { type, value } = props;

  if (type === 'command') {
    return <Text>Command Header Component</Text>;
  } else if (type === 'emoji') {
    return <Text>Emoji Header Component</Text>;
  }

  return <AutoCompleteSuggestionHeader type={type} value={value} />;
};

const App = () => {
  return (
    <Channel AutoCompleteSuggestionHeader={MyCustomHeader}>
      {/* The underlying components */}
    </Channel>
  );
};
```

Note: The default `AutoCompleteSuggestionHeader` should be returned by default if you don't want customized implementations for all trigger types.

### Customizing Item Component

The AutoCompleteSuggestionItem component can be customized in the same way as the Header above. To customize it, pass your component with the `AutoCompleteSuggestionItem` prop on a Channel component.

The props for the component are as follows:

- `type`: The trigger type of the suggestion list
- `itemProps`: Each trigger type has a different set of values passed in itemProps. See the following table for the specific values:

| Trigger type | `itemProps` value                                                                                                                                                      |
|--------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `command`    | `name`: The name of the command to trigger<br />`args`: Arguments passed to the command                                                                                |
| `emoji`      | `name`: The name of the emoji to render<br />`unicode`: The unicode codepoints of the emoji                                                                            |
| `mention`    | See the type [`SuggestionUser`](https://github.com/GetStream/stream-chat-react-native/blob/develop/package/src/contexts/suggestionsContext/SuggestionsContext.tsx#L43) |


```jsx
import { Text, View } from 'react-native';
import {
  AutoCompleteSuggestionItem,
  AutoCompleteSuggestionItemProps,
  Avatar,
  Channel,
  CommandItemType,
} from 'stream-chat-react-native';

const MyCustomItem = (props: AutoCompleteSuggestionItemProps) => {
  const { itemProps, type } = props;

  if (type === 'command') {
    const { name, args } = itemProps as CommandItemType;
    return (
      <View>
        <Text>{name}</Text>
        <Text>{args}</Text>
      </View>
    );
  } else if (type === 'mention') {
    const { id, image, name, online } = itemProps as SuggestionUser<Us>;
    return (
      <View style={{ flexDirection: 'row' }}>
        <Avatar image={image} name={name} online={online} size={avatarSize} />
        <Text>{itemProps.name}</Text>
      </View>
    );
  }

  return <AutoCompleteSuggestionItem itemProps={itemProps} type={type} />;
};

const App = () => {
  return (
    <Channel AutoCompleteSuggestionItem={MyCustomItem}>{/* The underlying components */}</Channel>
  );
};
```

Note: The default `AutoCompleteSuggestionItem` must be returned if you don't want any change for all trigger types.

### Customizing the AutoComplete suggestion list

The list can also be customized in its entirety with its own header, item and any other changes you may want. You can do this by passing your own component in the `AutoCompleteSuggestionList` prop on a Channel component.

The props available to the component are:
- `active`: Checks the list is active or not.
- `headerProps`: The props for the header of the list. This can take `type` and `value` as explained above in the header section.
- `suggestions`: The suggestions array which contains the data for the list to be displayed in the list items.
- `type`: The trigger type of the suggestion list.

An example to demonstrate the same is as follows. The different ways of usage is also explained:

```jsx
import {
  AutoCompleteSuggestionHeader,
  AutoCompleteSuggestionItem,
  Channel,
} from 'stream-chat-react-native';
import { FlatList, Text, View } from 'react-native';

const MyCustomList = (props: AutoCompleteSuggestionListProps) => {
  const {
    headerProps: { value },
    suggestions: { query, data, onSelect },
    type,
  } = props;

  if (type === 'command') {
    return (
      <View>
        <View>
          <Text>Command suggestions for {value}</Text>
        </View>
        {data.map((item) => (
          <AutoCompleteSuggestionItem itemProps={item} type={type} />
        ))}
      </View>
    );
  } else if (type === 'emoji') {
    return (
      <FlatList
        data={data}
        keyboardShouldPersistTaps='always'
        ListHeaderComponent={<AutoCompleteSuggestionHeader type={type} value={value} />}
        renderItem={({ index, item }) => {
          return (
            <TouchableOpacity
              onPress={() => {
                onSelect(item);
              }}
            >
              <Text>{item.name}</Text>
            </TouchableOpacity>
          );
        }}
      />
    );
  }

  return (
    <View>
      <AutoCompleteSuggestionHeader type={type} value={query} />
      {data.map((item) => (
        <AutoCompleteSuggestionItem itemProps={item} type={type} />
      ))}
    </View>
  );
};

const App = () => {
  return (
    <Channel AutoCompleteSuggestionList={MyCustomList}>{/* The underlying components */}</Channel>
  );
};
```

Note: The default `AutoCompleteSuggestionList` must be returned if you don't want any change for a particular trigger type.

````

I've made some suggestions on how to improve the documentation a bit, and moved the customized components to be components there too since that's how we're talking about it in the documentation.